File: snippets/csharp/VS_Snippets_ADO.NET/Classic WebData ConstraintCollection Example/CS/source.cs
using System;
using System.Data;
using System.Windows.Forms;

public class Form1: Form
{
    protected DataSet DataSet1;


    // <Snippet1>
    private void MakeTableWithUniqueConstraint()
    {
        var table = new DataTable("table");
        var column = new DataColumn("UniqueColumn");
        column.Unique = true;
        table.Columns.Add(column);

        // Print count, name, and type.
        Console.WriteLine($"Constraints.Count {table.Constraints.Count}");
        Console.WriteLine(table.Constraints[0].ConstraintName);
        Console.WriteLine(table.Constraints[0].GetType() );

        // Add a second unique column.
        column = new DataColumn("UniqueColumn2");
        column.Unique = true;
        table.Columns.Add(column);

        // Print info again.
        Console.WriteLine($"Constraints.Count {table.Constraints.Count}");
        Console.WriteLine(table.Constraints[1].ConstraintName);
        Console.WriteLine(table.Constraints[1].GetType() );
    }
     
    private void MakeTableWithForeignConstraint()
    {
        // Create a DataSet.
        var dataSet = new DataSet("dataSet");

        // Make two tables.
        var customersTable = new DataTable("Customers");
        var ordersTable = new DataTable("Orders");

        // Create four columns, two for each table.
        var name = new DataColumn("Name");
        var id = new DataColumn("ID");
        var orderId = new DataColumn("OrderID");
        var cDate = new DataColumn("OrderDate");
 
        // Add columns to tables.
        customersTable.Columns.Add(name);
        customersTable.Columns.Add(id);
        ordersTable.Columns.Add(orderId);
        ordersTable.Columns.Add(cDate);
 
        // Add tables to the DataSet.
        dataSet.Tables.Add(customersTable);
        dataSet.Tables.Add(ordersTable); 

        // Create a DataRelation for two of the columns.
        var myRelation = new DataRelation("CustomersOrders", id, orderId, true);
        dataSet.Relations.Add(myRelation);

        // Print TableName, Constraints.Count, 
        // ConstraintName and Type.
        foreach (DataTable t in dataSet.Tables)
        {
            Console.WriteLine(t.TableName);
            Console.WriteLine($"Constraints.Count {t.Constraints.Count}");
            Console.WriteLine($"ParentRelations.Count {t.ParentRelations.Count}");
            Console.WriteLine($"ChildRelations.Count {t.ChildRelations.Count}");
            foreach (Constraint cstrnt in t.Constraints)
            {
                Console.WriteLine(cstrnt.ConstraintName);
                Console.WriteLine(cstrnt.GetType());
            }
        }
    }
    // </Snippet1>

}

File: snippets/visualbasic/VS_Snippets_ADO.NET/Classic WebData Constraint Example/VB/source.vb
Imports System
Imports System.Data
Imports System.Windows.Forms

Public Class Form1
    Inherits Form
    Protected DataSet1 As DataSet  
    
' <Snippet1>
    Private Sub GetConstraints(dataTable As DataTable)
        Console.WriteLine()

        ' Print the table's name.
        Console.WriteLine($"TableName: {dataTable.TableName}")

        ' Iterate through the collection and print 
        ' each name and type value.
        Dim constraint As Constraint
        For Each constraint In  dataTable.Constraints
            Console.WriteLine($"Constraint Name: {constraint.ConstraintName}")
            Console.WriteLine($"Type: {constraint.GetType().ToString()"})

            ' If the constraint is a UniqueConstraint, 
            ' print its properties using a function below.
            If TypeOf constraint Is UniqueConstraint Then
                PrintUniqueConstraintProperties(constraint)
            End If

            ' If the constraint is a ForeignKeyConstraint, 
            ' print its properties using a function below.
            If TypeOf constraint Is ForeignKeyConstraint Then
                PrintForeigKeyConstraintProperties(constraint)
            End If
        Next constraint
    End Sub
    
    Private Sub PrintUniqueConstraintProperties( _
        constraint As Constraint)

        Dim uniqueConstraint = CType(constraint, UniqueConstraint)

        ' Get the Columns as an array.
        Dim columnArray As DataColumn() uniqueConstraint.Columns

        ' Print each column's name.
        For i As Integer = 0 To columnArray.Length - 1
            Console.WriteLine($"Column Name: {columnArray(i).ColumnName}")
        Next
    End Sub
    
    Private Sub PrintForeigKeyConstraintProperties( _
        constraint As Constraint)

        Dim fkConstraint = CType(constraint, ForeignKeyConstraint)
        
        ' Get the Columns as an array.
        Dim columnArray As DataColumn() fkConstraint.Columns
        
        ' Print each column's name.
        For i As Integer = 0 To columnArray.Length - 1
            Console.WriteLine($"Column Name: {columnArray(i).ColumnName}")
        Next
        Console.WriteLine()
        
        ' Get the related columns and print each columns name.
        columnArray = fkConstraint.RelatedColumns
        For i As Integer = 0 To columnArray.Length - 1
            Console.WriteLine($"Related Column Name: {columnArray(i).ColumnName}")
        Next
        Console.WriteLine()
    End Sub
' </Snippet1>
End Class
